src/blog/2024/09/node-red-version-control-with-snapshots.md

```diff
@@ -0,0 +1,246 @@
+*Image showing the snapshot view window in FlowFuse.*
+
+2. After clicking, a new window will open displaying the entire flow for that snapshot. At the bottom right of this window, you'll find options to **copy** or **download** the flow.
+
+#### Comparing Snapshots:
+
+If you have multiple snapshots and want to compare them, follow these steps:
+
+![Image showing the 'Compare Snapshot' option in the menu](./images/compare-snapshot-option.png){data-zoomable}
+*Image showing the 'Compare Snapshot' option in the menu*
+
+1. Click on the three-dot icon next to the snapshot you want to compare, then choose **Compare Snapshot**.
+
+![Image showing the comparison window in FlowFuse, featuring a dropdown to select the instance to compare and a compare button.](./images/window-comparing-snapshots.png){data-zoomable}
+*Image showing the comparison window in FlowFuse, featuring a dropdown to select the instance to compare and a compare button*
+
+2. In the newly opened window, select another snapshot from the dropdown to compare it with the one you chose.
+
+![Image showing the comparison window for snapshots in FlowFuse, with differences highlighted in light purple color.](./images/window-comparing-snapshots-2.png){data-zoomable}
+*Image showing the comparison window for snapshots in FlowFuse.*
+
+
+3. The flows of both snapshots will be displayed side by side. At the top-right corner of the window, you will see two buttons, Use according to your preference:
+   - **Prev**: Navigate to the previous difference between the two snapshots.
+   - **Next**: Navigate to the next difference.
+
+This comparison feature helps you easily identify changes, making it simple to spot issues or track flow modifications between versions.
+
+### Downloading, Uploading, and Deploying Snapshots
+
+Created snapshots can be downloaded locally, providing you with a backup of your Node-RED flows. This backup can be uploaded to another instance or restored in case of emergencies.
+
+#### Downloading Snapshot:
+
+![Image showing the 'Download Snapshot' option in the menu](./images/download-snapshot.png){data-zoomable}
+*Image showing the 'Download Snapshot' option in the menu*
+
+1. Click on the **three-dot** icon next to the corresponding snapshot, then select **Download Snapshot** from the menu.
+
+![Image showing the prompt to download a snapshot in FlowFuse, featuring a highlighted key and the download button.](./images/download-snapshot-prompt.png){data-zoomable}
+*Image showing the prompt to download a snapshot in FlowFuse, featuring a highlighted key and the download button*
+
+2. A **Download Snapshot** dialog will open, displaying a **secret key** if your snapshot flow has config nodes otherwise it not be displayed copy it if it is avialbe. This is used to encrypt sensitive info in the snapshot. As a convenience, a random secret is auto generated however you should change this to something memorable as it will be needed when you need to upload the snapshot.
```

Comment:
```suggestion
2. A **Download Snapshot** dialog will open.  Note: If the flows contain sensitive values, an additional field for **secret key** will be show. This is used to encrypt the sensitive values in the snapshot. As a convenience, a random secret is auto generated however you should change this to something memorable as it will be needed when you later upload the snapshot.
```